connector/src/main/scala/quasar/qscript/Transform.scala

```diff
@@ -86,9 +86,11 @@ class Transform
 
     (left.value, right.value) match {
       case (Embed(QC(Unreferenced())), r) =>
-        AutoJoinResult(AutoJoinBase(r, rann.provenance), lann.values, rann.values)
+        val buckets = prov.joinProvenances(List(provenance.Nada()), rann.provenance)
```

Comment:
Interesting. I figured it didn’t matter, since there was no provenance. But, rather than `List(Nada())`, try `lann.provenance`. It should have the same result, but `lann.provenance` is currently `Nil` and `joinProvenance` pads the shorter list with `Nada`.

So … using `Nil` instead of `List(Nada)` means that if `rann` has a provenance of `Nil` (like, it’s been reduced down to a single value), we won’t end up padding it with an extra `Both(Nada, Nada)`. And using `lann.provenance` means that if we _somehow_ end up with an unreferenced src with non-empty provenance (perhaps with constant provenance?), then we won’t accidentally drop it.

I’m happy to make this change, and thanks for figuring it out. So, you can either update, or I’ll merge and make this change on my original PR.